File: code/backend/src/main/java/org/example/backend/service/BusinessModuleService.java
package org.example.backend.service;

import org.example.backend.dto.BusinessModuleRequest;
import org.example.backend.dto.BusinessModuleResponse;
import java.util.List;

public interface BusinessModuleService {
    BusinessModuleResponse createModule(Long projectId, BusinessModuleRequest request, Long userId);
    BusinessModuleResponse updateModule(Long id, Long projectId, BusinessModuleRequest request, Long userId);
    void deleteModule(Long id, Long projectId, Long userId);
    List<BusinessModuleResponse> getModulesByProject(Long projectId);
    BusinessModuleResponse assignMember(Long id, Long projectId, Long assigneeId, Long userId);
}

File: code/frontend/src/features/requirement/services/businessModuleService.js
import axiosInstance from '@/api/axiosConfig';

const BASE_URL = '/projects';

export const businessModuleService = {
  getModulesByProject: async (projectId) => {
    const response = await axiosInstance.get(`${BASE_URL}/${projectId}/modules`);
    return response.data.data; // ApiResponse format
  },

  createModule: async (projectId, moduleData) => {
    const response = await axiosInstance.post(`${BASE_URL}/${projectId}/modules`, moduleData);
    return response.data.data;
  },

  updateModule: async (projectId, moduleId, moduleData) => {
    const response = await axiosInstance.put(`${BASE_URL}/${projectId}/modules/${moduleId}`, moduleData);
    return response.data.data;
  },

  deleteModule: async (projectId, moduleId) => {
    const response = await axiosInstance.delete(`${BASE_URL}/${projectId}/modules/${moduleId}`);
    return response.data.data;
  },

  assignMember: async (projectId, moduleId, assigneeId) => {
    const response = await axiosInstance.patch(`${BASE_URL}/${projectId}/modules/${moduleId}/assignee`, { assigneeId });
    return response.data.data;
  }
};
